src/DefaultFileSystem.cpp

```diff
@@ -78,8 +78,23 @@ DefaultFileSystem::Read(const std::string& path) const
 void DefaultFileSystem::Write(const std::string& path,
                               std::tr1::shared_ptr<std::istream> data)
 {
-  std::ofstream file(NormalizePath(path).c_str(), std::ios_base::out | std::ios_base::binary);
-  file << Utils::Slurp(*data);
+  std::wstring pathnormalized = NormalizePath(path);
+  int poslastpoint = pathnormalized.find_last_of(L".");
+  std::wstring filename = pathnormalized.substr(0, poslastpoint - 1);
+  std::wstring tempfile = filename + L".tmp";
+  std::ofstream file(tempfile.c_str(), std::ios_base::out | std::ios_base::binary);
+  if (file.good())
+  {
+    file << Utils::Slurp(*data);
+    if (file.good())
+    {
+      file.close();
+      if (file.good())
+      {
+        rename(tempfile.c_str(), pathnormalized.c_str());
```

Comment:
I think we need something to reduce `std::string <> std::wstring` indirection here and we should not create a code duplication.